extensions/GitExtension/FileExplorerGitIntegration.UnitTest/WslIntegratorUnitTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using FileExplorerGitIntegration.Models;
+
+namespace FileExplorerGitIntegration.UnitTest;
+
+[TestClass]
+public class WslIntegratorUnitTests
+{
+    [TestMethod]
+    [DataRow("\\wsl$\\Ubuntu-20.04\\home\\user\\repo")]
+    [DataRow("\\wsl.localhost\\Ubuntu-20.04\\home\\user\\repo")]
+    [DataRow("\\wsl$\\Ubuntu\\home\\user\\repo")]
+    [DataRow("\\wsl.localhost\\Ubuntu\\home\\user\\repo")]
+    [DataRow("\\wsl.localhost\\Debian\\home\\user\\repo")]
+    [DataRow("\\wsl$\\kali-linux\\home\\user\\repo")]
+    [DataRow("\\wsl$\\Ubuntu-18.04\\home\\user\\testRepo")]
+    [DataRow("\\wsl.localhost\\Ubuntu-18.04\\home\\user\\testRepo")]
+    public void IsWSLRepoPositiveTests(string repositoryPath)
+    {
+        Assert.IsTrue(WslIntegrator.IsWSLRepo(repositoryPath));
+    }
+
+    [TestMethod]
+    [DataRow("C:\\Users\\foo\\bar")]
+    [DataRow("\\wsl$*\\Ubuntu\\home\\user\\repo")]
+    [DataRow("D:\\wsl.localhost\\Ubuntu-20.04\\home\\user\\repo")]
+    [DataRow("\\wsl.test\\Ubuntu\\home\\user\\repo")]
+    public void IsWslRepoNegativeTests(string repositoryPath)
+    {
+        Assert.IsFalse(WslIntegrator.IsWSLRepo(repositoryPath));
+    }
+
+    [TestMethod]
+    [DataRow("\\wsl$\\Ubuntu-20.04\\home\\user\\repo", "Ubuntu-20.04")]
```

Comment:
Validate that if we pass in a non-WSL path, or an incorrectly formed WSL path, we reject it and don't return a distro name.

Reply:
Made test modifications in latest changes. Thanks!